src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -164,41 +170,44 @@ private void VerifyConnected()
 				// need to try IP Addresses one at a time: https://github.com/dotnet/corefx/issues/5829
 				foreach (var ipAddress in ipAddresses)
 				{
-					cancellationToken.ThrowIfCancellationRequested();
-
-					Socket socket = null;
+					var socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
 					try
 					{
-						socket = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
-						using (cancellationToken.Register(() => socket.Dispose()))
+						var connectTasks = new List<Task>
```

Comment:
The Socket tasks do not take a cancellation token as an argument, so the Timeout Cancellation token will not do anything.  In order to find a timeout, I have leveraged Task.WaitAny with a timeout task, and checked the socket status.